maven-model-builder/src/test/java/org/apache/maven/model/profile/activation/FileProfileActivatorTest.java

```diff
@@ -0,0 +1,144 @@
+                new ProfileActivationFilePathInterpolator().setPathTranslator( new DefaultPathTranslator() ) );
+
+        context.setProjectDirectory( new File( PATH ) );
+
+        File file = new File( PATH + FILE );
+        if ( !file.exists() )
+        {
+            if ( !file.createNewFile() )
+            {
+                throw new RuntimeException( String.format( "Can't create %s", file ) );
+            }
+        }
+    }
+
+
+    public void test_isActive_noFile()
+    {
+        assertActivation( false, newExistsProfile( null ), context );
+        assertActivation( false, newExistsProfile( "someFile.txt" ), context );
+        assertActivation( false, newExistsProfile( "${basedir}/someFile.txt" ), context );
+
+        assertActivation( false, newMissingProfile( null ), context );
+        assertActivation( true, newMissingProfile( "someFile.txt" ), context );
+        assertActivation( true, newMissingProfile( "${basedir}/someFile.txt" ), context );
+    }
+
+    public void test_isActiveExists_fileExists()
+    {
+        assertActivation( true, newExistsProfile( FILE ), context );
+        assertActivation( true, newExistsProfile( "${basedir}" ), context );
+        assertActivation( true, newExistsProfile( "${basedir}/" + FILE ), context );
+
+        assertActivation( false, newMissingProfile( FILE ), context );
+        assertActivation( false, newMissingProfile( "${basedir}" ), context );
+        assertActivation( false, newMissingProfile( "${basedir}/" + FILE ), context );
+    }
+
+    public void test_isActiveExists_leavesFileUnchanged()
+    {
+        Profile profile = newExistsProfile( FILE );
+        assertEquals( profile.getActivation().getFile().getExists(), FILE );
+
+        assertActivation( true, profile, context );
+
+        assertEquals( profile.getActivation().getFile().getExists(), FILE );
+    }
+
+    private Profile newExistsProfile( String filePath )
+    {
+        return newProfile( filePath, true );
+    }
+
+    private Profile newMissingProfile( String filePath )
+    {
+        return newProfile( filePath, false );
+    }
+
+    private Profile newProfile( String filePath, boolean exists )
+    {
+        ActivationFile activationFile = new ActivationFile();
+        if ( exists )
+        {
+            activationFile.setExists( filePath );
+        }
+        else
+        {
+            activationFile.setMissing( filePath );
+        }
+
+        Activation activation = new Activation();
+        activation.setFile( activationFile );
+
+        Profile profile = new Profile();
+        profile.setActivation( activation );
+
+        return profile;
+    }
+
+    @Override
+    protected void tearDown() throws Exception
+    {
+        File file = new File( PATH + FILE );
+        if ( file.exists() )
+        {
+            if ( !file.delete() )
+            {
+                throw new RuntimeException( String.format( "Can't delete %s", file ) );
```

Comment:
not sure you need to fail here, but if you do an IOException matches the context

Reply:
I think it's a rare case when I can create a file before a test and can't remove it after. But when it happens, I'd like to attract attention. I replaced `RuntimeException` with `IOException`.